### PR TITLE
Add a Dockerfile

### PR DESCRIPTION
**Pull request by mahiso**
PR adds a Dockerfile to build docker image on (my) NAS or Docker hub.
No changes in other code, compose file or configuration.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] **I have read CONTRIBUTING.md**
- [x] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [ ] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [ ] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [ ] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [x] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

### CODE REVIEW

**Comment by rnevet** [comment]
@mahiso I'm missing the reason why this should be merged into the main repo.

**Comment by mahiso** [comment]
@rnevet It's only another way to create a docker image instead of using compose. I personally use it to build images automatically on docker.com via the docker/github integration.

You are right, it's not a must have or big reason. Therefore, not problem to close the PR.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM python:2.7-slim
+MAINTAINER Maik Hinrichs <maik@mahiso.de>
```
Comment (rnevet) [inline Dockerfile]: Your name and email?

**Comment by rnevet** [comment]
@mahiso I'm not against, just asking questions as I need to understand better the motivation.

File: Dockerfile (hunk 1 above)
Comment (mahiso) [inline Dockerfile]: Right. Why do you ask? Is there a project email address?

Comment (rnevet) [inline Dockerfile]: No, can you set it to GitHub url? or does it have to be an email?

Comment (mahiso) [inline Dockerfile]: I replaced it by a LABEL.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r ./requirements.txt
+
+COPY . .
+
+VOLUME /data
+
+RUN ln -s /data/market_data market_data; \
+    ln -s /data/log/botlog.json www/botlog.json
```
Comment (rnevet) [inline Dockerfile]: does this require specific configuration?

Comment (mahiso) [inline Dockerfile]: No, no configuration changes in config, web server or log json. All this are located in /data (could be a extra docker data volume or a local dir) and are safe from container changes (e.g remove and create a new one using new image version)
Directory structure in /data is:
/data
    |- marked_data
    |       | - marked data files
    |- log
    |       | - botlog.json
    |- config
    |        | - default.cfg

**Comment by rnevet** [comment]
@laxdog @Evanito any additional input?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Build: docker build -t <your_id>/pololendingbot .
+# Run: docker run -d -v /pololendingbot_data:/data -p 8000:8000 <your_id>/pololendingbot
+#
+
+WORKDIR /usr/src/app
+
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r ./requirements.txt
+
+COPY . .
```
Comment (laxdog) [inline Dockerfile]: Why not just copy everything to the workdir instead of messing about with symlinks and config paths.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+#
+# Build: docker build -t <your_id>/pololendingbot .
+# Run: docker run -d -v /pololendingbot_data:/data -p 8000:8000 <your_id>/pololendingbot
+#
+
+WORKDIR /usr/src/app
+
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r ./requirements.txt
```
Comment (laxdog) [inline Dockerfile]: Should this be CMD so the requirements get updated? Maybe not, I'm not sure. I do think all the RUNs should be on one line though to reduce the number of layers in the image.

Comment (mahiso) [inline Dockerfile]: Reason is that I dislike change dependencies after image creation during each start of the container. Never touch a running system...

File: Dockerfile (hunk 3 above)
Comment (mahiso) [inline Dockerfile]: Holding data in an extra volume protects it all for container changes. Never hold data in application container!

File: Dockerfile (hunk 4 above)
Comment (laxdog) [inline Dockerfile]: OK, fair enough.

File: Dockerfile (hunk 3 above)
Comment (laxdog) [inline Dockerfile]: OK. I'll merge this then.

**Pull request merged**